src/sql/workbench/api/node/mainThreadModelViewDialog.ts

```diff
@@ -32,6 +39,22 @@ export class MainThreadModelViewDialog implements MainThreadModelViewDialogShape
 		throw new Error('Method not implemented.');
 	}
 
+	public $openEditor(modelViewId: string, title: string, position?: vscode.ViewColumn): Thenable<void> {
+		return new Promise<void>((resolve, reject) => {
+			let input = new ModelViewInput(title, modelViewId);
+			let editorOptions = {
+				preserveFocus: true,
+				pinned: true
+			};
+
+			this._editorService.openEditor(input, editorOptions, position as any).then(() => {
+				resolve();
+			}, error => {
+				reject(error);
+			});
+		});
```

Comment:
What I meant with my previous comment is that this code should work but I think it can be simplified to the following without losing anything:

```
let input = new ModelViewInput(title, modelViewId);
let editorOptions = {
    preserveFocus: true,
    pinned: true
};
 
return this._editorService.openEditor(input, editorOptions, position as any).then(() => undefined);
```